simulator/src/main/java/com/hazelcast/simulator/worker/WorkerCommandRequestProcessor.java

```diff
@@ -353,18 +353,17 @@ private HazelcastInstance getHazelcastInstance() {
 
     private abstract class CommandThread extends Thread {
 
-        private final Command command;
         private final String testId;
 
         public CommandThread(Command command, String testId) {
-            this.command = command;
             this.testId = testId;
+
+            commands.put(testId, command);
```

Comment:
We should directly add the command to the map, which is used for the `IsPhaseCompletedCommand`, so don't have a race between the `CommandThread` setting this value in the `run()` method and the `TestCaseRunner` already asking for a phase completion.